### PR TITLE
feat: improve ICS misbehaviour E2E testing coverage

### PR DESCRIPTION
**Pull request by sainoe**
## Description

Closes: #1224

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

* [ ] Included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
* [ ] Added `!` to the type prefix if state-machine breaking change (i.e., requires coordinated upgrade)
* [ ] Confirmed this PR does not introduce changes requiring state migrations, OR migration code has been added to consumer and/or provider modules
* [ ] Targeted the correct branch (see [PR Targeting](https://github.com/cosmos/interchain-security/blob/main/CONTRIBUTING.md#pr-targeting))
* [ ] Provided a link to the relevant issue or specification
* [ ] Followed the guidelines for [building SDK modules](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/building-modules)
* [ ] Included the necessary unit and integration [tests](https://github.com/cosmos/interchain-security/blob/main/CONTRIBUTING.md#testing)
* [ ] Added a changelog entry to `CHANGELOG.md`
* [ ] Included comments for [documenting Go code](https://blog.golang.org/godoc)
* [ ] Updated the relevant documentation or specification
* [ ] Reviewed "Files changed" and left comments if necessary 
* [ ] Confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

* [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
* [ ] confirmed `!` in the type prefix if API or client breaking change
* [ ] confirmed this PR does not introduce changes requiring state migrations, OR confirmed migration code has been added to consumer and/or provider modules
* [ ] confirmed all author checklist items have been addressed
* [ ] reviewed state machine logic
* [ ] reviewed API design and naming
* [ ] reviewed documentation is accurate
* [ ] reviewed tests and test coverage

### CODE REVIEW

File: tests/e2e/steps_consumer_misbehaviour.go [hunk 1]
```diff
@@ -241,13 +241,15 @@ func stepsCauseConsumerMisbehaviour(consumerName string) []Step {
 						},
 					},
 				},
-				chainID(consumerName): ChainState{
```
Comment (sainoe) [inline tests/e2e/steps_consumer_misbehaviour.go:244]: Removed since our model state checker requires the chain to still produce blocks.

**Review by p-offtermatt — COMMENTED** [review]
The PR makes sense, but I am worried that we cannot distinguish between a chain halt and an unrelated panic - I'd suggest not reusing the waitBlocks function for this, and instead getting the current block twice with a 20 second delay and checking that the heights are the same.

**Comment by sainoe** [comment]
@p-offtermatt I agree with your concern. I will change this asap.

**Review by p-offtermatt — APPROVED** [review]
Thanks for addressing the concern, this looks good to me now!

**Pull request merged**